kafka.go

```diff
@@ -168,12 +168,7 @@ func (kafka *kafkaReader) Seek(ctx context.Context, offset int64) (int64, error)
 		kafka.closeAsync()
 	}
 
-	broker, err := kafka.client.Leader(kafka.topic, kafka.partition)
-	if err != nil {
-		return 0, err
-	}
-
-	offset, err = kafka.getOffset(broker, offset)
+	offset, err := kafka.getOffset(ctx, offset)
```

Comment:
`getOffset` here is used to translate the given offset `Seek(ctx, offset)` to a real kafka offset. If you have an offset like `515` then it won't actually make the call and just return the same offset. However, if the offset is `-1` or `-2` we need to ask Kafka to translate that into an actual offset to fetch. `-1` will cause Kafka to give us back the highest offset (high watermark) and `-2` will cause Kafka to give us back the lowest offset. 

However, I need to update the async task logic to move the offset updating logic to the `Read` method. We're only updating the offset after every batch, not every message. So if you do `Read()` then `Offset()` they won't match.